src/views/dashboard/users/AllUsers.vue

```diff
@@ -70,67 +81,176 @@
           show: false,
-          type: 'success',
+          text: '',
+          color: 'red',
         },
+        modalTextList: [],
+        timeout: 2000,
+        showEditModal: false,
+        modalFunction: '',
+        modalTitle: '',
+        modalButtonText: '',
+        userEdit: '',
       }
     },
     mounted () { this.loadData() },
     methods: {
+      showModal (us) {
+        this.modalTextList = [
+          { name: 'name', label: 'Name', value: us.name },
+          { name: 'email', label: 'E-mail', value: us.email },
+        ]
+        this.modalFunction = this.editUser
+        this.modalTitle = 'Change user information'
+        this.modalButtonText = 'Save'
+        this.showEditModal = true
+        this.userEdit = us
+      },
       loadData () {
         API.user.getAll()
           .then(response => {
             if ('error' in response) {
               this.alert = { show: true, text: 'Unable to fetch users:' + response.error, type: 'error' }
             } else {
               this.users = response.sort((a, b) => (a.modified_at < b.modified_at) ? 1 : -1)
-              this.alert = { show: true, text: 'Users loaded.', type: 'success' }
+              this.snack = { show: true, text: 'Users loaded.', color: 'success' }
             }
           })
       },
       deleteUser (i, user) {
         console.log('Deleting user with id: ' + user.id)
-        API.user.delete(user.email)
-          .then(() => {
-            this.users.splice(i, 1)
-            this.alert = { show: true, text: 'User ' + user.id + ' was deleted succesfully.', type: 'success' }
+        API.user.delete(user.id)
+          .then((response) => {
+            if ('error' in response) {
+              console.log(response)
+            } else {
+              this.users.splice(i, 1)
+              this.snack = { show: true, text: 'User ' + user.id + ' was deleted succesfully.', color: 'success' }
+            }
           })
           .catch((error) => {
             console.log(error)
-            this.alert = { show: true, text: 'There was an error deleting the user ' + user.id + '.', type: 'error' }
+            this.snack = { show: true, text: 'There was an error deleting the user ' + user.id + '.', color: 'error' }
           })
       },
-      editUser (i, user) {
-        console.log('Editing user with id: ' + user.id)
+      editUser (payload) {
+        console.log('Editing user with id: ' + this.userEdit.id)
+        console.log(payload)
+        this.userEdit.name = payload.name.value
+        this.userEdit.email = payload.email.value
+        // meter correo, nombre y revisar qué más
+        // quitar admin y hacerlo con switch
+        // mostrar de otra manera el superadmin
+        this.showEditModal = false
+        console.log(this.userEdit)
+        API.user.put(
+          this.userEdit.id,
+          { name: this.userEdit.name, email: this.userEdit.email },
+          { 'Content-Type': 'application/json' },
```

Comment:
You could put this as a default value for the `headers` argument. That way you would not to fill it at each time.

Reply:
I don't know if it's the correct solution, but I added this { 'Content-Type': 'application/json' } inside the headers argument in the makeresource.js (create and put functions)